pkg/providers/vsphere/vsphere_test.go

```diff
@@ -3151,6 +3139,37 @@ func TestProviderGenerateCAPISpecForCreateWithPodIAMConfig(t *testing.T) {
 	test.AssertContentToFile(t, string(cp), "testdata/expected_results_pod_iam_config.yaml")
 }
 
+func TestProviderGenerateCAPISpecForCreateWithMultipleMachineTemplate(t *testing.T) {
```

Comment:
instead of creating a new test and adding 2 long test files, can we change an existing test and make them use different template and test this change?

Reply:
I basically use `cluster_main.yaml`but it's also referenced at other places so think better to have a separate test file for this case